src/cpp/include/openvino/genai/streamer_base.hpp

```diff
@@ -4,16 +4,28 @@
 #pragma once
 
 #include "openvino/genai/tokenizer.hpp"
+#include <variant>
 
 namespace ov {
 namespace genai {
 
+enum class StreamerRunningStatus {
+    UNDEF = 0, // Streaming is not run
+    RUNNING = 1, // Continue to run of inference
```

Comment:
RUNNING and UNDEF seem to be equivalent. In that case you should keep only one of them. Moreover callback should never return UNDEF, so merging them fixes the API.

Reply:
removed it , moved to GenerationStatus

Reply:
Merging with `GenerationStatus` allows a callback to return FINISHED and IGNORED which aren't related to this. I'd guess https://github.com/openvinotoolkit/openvino.genai/pull/1476#discussion_r1909969883 was about aligning API but not merging. @ilya-lavrenov, is that so?

Reply:
yes

Reply:
updated